cmd/server/cadence.go

```diff
@@ -72,18 +75,37 @@ func startHandler(c *cli.Context) {
 		log.Fatal("Incompatible sql versions: ", err)
 	}
 
+	var daemons []common.Daemon
 	services := getServices(c)
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
 	for _, svc := range services {
 		if _, ok := cfg.Services[svc]; !ok {
 			log.Fatalf("`%v` service missing config", svc)
 		}
 		server := newServer(svc, &cfg)
+		daemons = append(daemons, server)
 		server.Start()
 	}
 
+	go shutdownDaemonsOnSignal(sigc, daemons)
+
 	select {}
```

Comment:
Just a suggestion - instead of spinning up a new goroutine - consider renaming your shutdownDaemonsOnSignal to something like blockOnSigTerm - and remove the select{} on this method. The select{} exist only because we don't have something like a sig handler.

Reply:
Addressed